### PR TITLE
[UX2.0] Add OSPF & OSPFv3 Profile Parcels

### PR DESCRIPTION
**Pull request by seconroy**
### Description

Adds the following Profile Parcel resources and data sources:
- Transport Routing/OSPF
- Transport Routing/OSPFv3/IPv4
- Transport Routing/OSPFv3/IPv6
- Service Routing/OSPF
- Service Routing/OSPFv3/IPv4
- Service Routing/OSPFv3/IPv6

### Types of Changes

* [x] New feature (non-breaking change which adds functionaality)
* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Build/CI change
* [ ] Code quality improvement/refactoring/documentation (no functional changes)

### Checklist

* [x] My code follows the code style of this project
* [x] I have added tests to cover my changes
* [x] All new and existing tests pass locally

### CODE REVIEW

File: gen/definitions/profile_parcels/service_routing_ospf.yaml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        example: 10
+      - model_name: priority
+        tf_name: designated_router_priority
+        example: 1
+      - model_name: network
+        tf_name: ospf_network_type
+        example: broadcast
+      - model_name: passiveInterface
+        example: false
+      - model_name: type
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospf.yaml:109]: tf_name: authentication_type

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    example: f6dd22c8-0b4f-496c-9a0b-6813d1f8b8ac
+    test_value: sdwan_service_feature_profile.test.id
+  - model_name: routerId
+    data_path: [basic]
+    example: 1.2.3.4
+  - model_name: distance
+    data_path: [basic]
+    example: 110
+  - model_name: externalDistance
+    tf_name: distance_for_external_routes
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:22]: "distance_external"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [basic]
+    example: 110
+  - model_name: interAreaDistance
+    tf_name: distance_for_inter_area_routes
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:26]: distance_inter_area

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [basic]
+    example: 110
+  - model_name: intraAreaDistance
+    tf_name: distance_for_intra_area_routes
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:30]: distance_intra_area

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [basic]
+    example: 110
+  - model_name: referenceBandwidth
+    data_path: [advanced]
+    example: 101
+  - model_name: compatibleRfc1583
+    tf_name: rfc_1583_compatible
+    data_path: [advanced]
+    example: true
+  - model_name: originate
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:40]: Would add "tf_name: default_information_originate"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [advanced, defaultOriginate]
+    example: false
+  - model_name: always
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:43]: Would add "tf_name: default_information_originate_always"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [advanced, defaultOriginate]
+    example: false
+  - model_name: metric
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:46]: Would add "tf_name: default_information_originate_metric"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [advanced, defaultOriginate]
+    example: 1
+  - model_name: metricType
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:49]: Would add "tf_name: default_information_originate_metric_type"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [advanced, defaultOriginate]
+    example: type1
+  - model_name: delay
+    tf_name: spf_calculation_deplay
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:53]: typo, should be "spf_calculation_delay"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [advanced, spfTimers]
+    example: 200
+  - model_name: initialHold
+    tf_name: initial_hold_time
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:57]: spf_initial_hold_time

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [advanced, spfTimers]
+    example: 1000
+  - model_name: maxHold
+    tf_name: maximum_hold_time
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:61]: spf_maximum_hold_time

File: gen/definitions/profile_parcels/service_routing_ospf.yaml [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+  - model_name: maxHold
+    tf_name: maximum_hold_time
+    example: 10000
+  - model_name: redistribute
+    tf_name: redistributes
+    attributes: 
+    - model_name: protocol
+      id: true
+      example: static
+    - model_name: dia
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospf.yaml:53]: tf_name: nat_dia

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+    data_path: [advanced]
+    example: false
+  - model_name: redistribute
+    tf_name: redistributes
+    attributes: 
+    - model_name: protocol
+      id: true
+      example: nat-route
+    - model_name: natDia
+      tf_name: enable_nat_dia
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:79]: Change here to "nat_dia" to have it in line with OSPFv2

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+      tf_name: route_policy_id
+      data_path: [routePolicy]
+      exclude_test: true
+      example: 615d948f-34ee-4a2e-810e-a9bd8d3d48ec
+  - model_name: action
+    tf_name: router_lsa_action
+    data_path: [maxMetricRouterLsa]
+    example: on-startup
+  - model_name: onStartUpTime
+    tf_name: router_lsa_on_startu_p_time
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:91]: typo, should be router_lsa_on_startup_time

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+      data_path: [areaTypeConfig]
+      exclude_test: true
+      example: false
+    - model_name: alwaysTranslate
+      data_path: [areaTypeConfig]
+      exclude_test: true
+      example: false
+    - model_name: interfaces
+      attributes:
+      - model_name: ifName
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:116]: tf_name: name

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+        example: GigabitEthernet2
+      - model_name: helloInterval
+        example: 10
+      - model_name: deadInterval
+        example: 40
+      - model_name: retransmitInterval  
+        tf_name: lsa_retransmit_interval
+        example: 5
+      - model_name: cost
+        tf_name: interface_cost
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:127]: we could remove that and leave it as "cost"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+        example: 10
+      - model_name: networkType
+        tf_name: ospf_network_type
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:130]: "network_type"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+        example: broadcast
+      - model_name: passiveInterface
+        example: false
+      - model_name: authType
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:134]: tf_name: authentication_type

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 19]
```diff
@@ -0,0 +1,163 @@
...
+        data_path: [authenticationConfig]
+        example: no-auth
+      - model_name: spi
+        data_path: [authenticationConfig]
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:138]: tf_name: authentication_spi

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml [hunk 20]
```diff
@@ -0,0 +1,163 @@
...
+        exclude_test: true
+        example: 12345
+      - model_name: authKey
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv4.yaml:141]: tf_name: authentication_key

File: gen/definitions/profile_parcels/service_routing_ospf.yaml [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+        example: 7
+      - model_name: md5
+        tf_name: message_digest_key
+        write_only: true
+        example: sdjfhsghbjdjr
+    - model_name: range
+      tf_name: ranges
+      attributes: 
+      - model_name: ipAddress
+        data_path: [address]
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospf.yaml:122]: In OSPFv3 IPv4 we just have "address/mask" here we have "ipAddress/subnetMask" - we should align this in both, not sure which option is better

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 22]
```diff
@@ -0,0 +1,156 @@
...
+    example: f6dd22c8-0b4f-496c-9a0b-6813d1f8b8ac
+    test_value: sdwan_service_feature_profile.test.id
+  - model_name: routerId
+    data_path: [basic]
+    example: 1.2.3.4
+  - model_name: distance
+    data_path: [basic]
+    example: 110
+  - model_name: externalDistance
+    tf_name: distance_for_external_routes
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:22]: distance_external

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 23]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [basic]
+    example: 110
+  - model_name: interAreaDistance
+    tf_name: distance_for_inter_area_routes
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:26]: distance_inter_area

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 24]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [basic]
+    example: 110
+  - model_name: intraAreaDistance
+    tf_name: distance_for_intra_area_routes
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:30]: distance_intra_area

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 25]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [basic]
+    example: 110
+  - model_name: referenceBandwidth
+    data_path: [advanced]
+    example: 101
+  - model_name: compatibleRfc1583
+    tf_name: rfc_1583_compatible
+    data_path: [advanced]
+    example: true
+  - model_name: originate
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:40]: Would add "tf_name: default_information_originate"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 26]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [advanced, defaultOriginate]
+    example: false
+  - model_name: always
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:43]: Would add "tf_name: default_information_originate_always"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 27]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [advanced, defaultOriginate]
+    example: false
+  - model_name: metric
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:46]: Would add "tf_name: default_information_originate_metric"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 28]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [advanced, defaultOriginate]
+    example: 1
+  - model_name: metricType
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:49]: Would add "tf_name: default_information_originate_metric_type"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 29]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [advanced, defaultOriginate]
+    example: type1
+  - model_name: delay
+    tf_name: spf_calculation_deplay
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:53]: typo, should be "spf_calculation_delay"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 30]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [advanced, spfTimers]
+    example: 200
+  - model_name: initialHold
+    tf_name: initial_hold_time
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:57]: spf_initial_hold_time

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 31]
```diff
@@ -0,0 +1,156 @@
...
+    data_path: [advanced, spfTimers]
+    example: 1000
+  - model_name: maxHold
+    tf_name: maximum_hold_time
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:61]: spf_maximum_hold_time

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 32]
```diff
@@ -0,0 +1,156 @@
...
+      tf_name: route_policy_id
+      data_path: [routePolicy]
+      exclude_test: true
+      example: 615d948f-34ee-4a2e-810e-a9bd8d3d48ec
+  - model_name: action
+    tf_name: router_lsa_action
+    data_path: [maxMetricRouterLsa]
+    example: on-startup
+  - model_name: onStartUpTime
+    tf_name: router_lsa_on_startu_p_time
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:88]: typo, should be router_lsa_on_startup_time

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 33]
```diff
@@ -0,0 +1,156 @@
...
+      data_path: [areaTypeConfig]
+      exclude_test: true
+      example: false
+    - model_name: alwaysTranslate
+      data_path: [areaTypeConfig]
+      exclude_test: true
+      example: false
+    - model_name: interfaces
+      attributes:
+      - model_name: ifName
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:113]: tf_name: name

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 34]
```diff
@@ -0,0 +1,156 @@
...
+        example: GigabitEthernet2
+      - model_name: helloInterval
+        example: 10
+      - model_name: deadInterval
+        example: 40
+      - model_name: retransmitInterval  
+        tf_name: lsa_retransmit_interval
+        example: 5
+      - model_name: cost
+        tf_name: interface_cost
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:124]: we could remove that and leave it as "cost"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 35]
```diff
@@ -0,0 +1,156 @@
...
+        example: 10
+      - model_name: networkType
+        tf_name: ospf_network_type
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:127]: "network_type"

File: gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml [hunk 36]
```diff
@@ -0,0 +1,156 @@
...
+        example: broadcast
+      - model_name: passiveInterface
+        example: false
+      - model_name: authType
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/service_routing_ospfv3_ipv6.yaml:131]: tf_name: authentication_type

File: gen/definitions/profile_parcels/transport_routing_ospf.yaml [hunk 37]
```diff
@@ -0,0 +1,137 @@
...
+      example: true
+    - model_name: refId
+      tf_name: route_policy_id
+      data_path: [routePolicy]
+      exclude_test: true
+      example: 615d948f-34ee-4a2e-810e-a9bd8d3d48ec
+  - model_name: routerLsa
+    tf_name: router_lsas
+    attributes: 
+    - model_name: adType
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/transport_routing_ospf.yaml:63]: tf_name: type

File: gen/definitions/profile_parcels/transport_routing_ospfv3_ipv4.yaml [hunk 38]
```diff
@@ -0,0 +1,163 @@
...
+        example: GigabitEthernet2
+      - model_name: helloInterval
+        example: 10
+      - model_name: deadInterval
+        example: 40
+      - model_name: retransmitInterval  
+        tf_name: lsa_retransmit_interval
+        example: 5
+      - model_name: cost
+        tf_name: interface_cost
```
Comment (tzarski0) [inline gen/definitions/profile_parcels/transport_routing_ospfv3_ipv4.yaml:127]: we could remove that and leave it as "cost"

**Review by tzarski0 — CHANGES_REQUESTED** [review]

**Review by danischm — APPROVED** [review]

**Pull request merged**